server/routes/opensearch.ts

```diff
@@ -350,12 +405,34 @@ export default class OpenSearchService {
         this.client
       );
 
-      const response = await callWithRequest(
-        'indices.getMapping', {
-          index,
+      let mappings: Mappings = {};
+      let remoteMappings: Mappings = {};
+      let localIndices: string[] = indices.filter(
+        (index: string) => !index.includes(':')
+      );
+      let remoteIndices: string[] = indices.filter((index: string) =>
+        index.includes(':')
+      );
+
+      if (localIndices.length > 0) {
+        mappings = await callWithRequest('indices.getMapping', {
+          index: localIndices,
+        });
+      }
+
+      // make call to fields_caps
+      if (remoteIndices.length) {
+        const fieldCapsResponse = await callWithRequest('transport.request', {
+          method: 'GET',
+          path:
+            remoteIndices.toString() + '/_field_caps?fields=*&include_unmapped',
         });
+        remoteMappings = convertFieldCapsToMappingStructure(fieldCapsResponse);
+      }
+      Object.assign(mappings, remoteMappings);
```

Comment:
Does this Object.assign works fine on merging nested structure? Should we create a new map object instead of mutating the `mappings` variable directly?

Reply:
I was reading online that Object.assign is more efficient which can be important as we can have indices with very large mappings. I tested it on complicated nested structures